api/node/index.ts

```diff
@@ -438,12 +440,68 @@ export function loadFile(filePath: string, options?: LoadFileOptions): Object {
     return slint_module;
 }
 
-// This api will be removed after teh event loop handling is merged check PR #3718.
-// After that this in no longer necessary.
-export namespace Timer {
-    export function singleShot(duration: number, handler: () => void) {
-        napi.singleshotTimer(duration, handler)
+class EventLoop {
+    #quit_loop: boolean = false;
+    #termination_promise: Promise<unknown> | null = null;
+    #terminate_resolve_fn: ((_value: unknown) => void) | null;
+    constructor() {
+    }
+
+    start(running_callback?: Function) {
+        if (this.#termination_promise != null) {
+            return this.#termination_promise;
+        }
+
+        this.#termination_promise = new Promise((resolve) => {
+            this.#terminate_resolve_fn = resolve;
+        });
+        this.#quit_loop = false;
+
+        if (running_callback! != undefined) {
+            napi.invokeFromEventLoop(() => {
+                running_callback();
+                running_callback = undefined;
+            });
+        }
+
+        let id = setInterval(() => {
+            if (napi.processEvents() == napi.ProcessEventsResult.Exited || this.#quit_loop) {
+                clearInterval(id);
+                this.#terminate_resolve_fn!(undefined);
+                this.#terminate_resolve_fn = null;
+                this.#termination_promise = null;
+                return;
+            }
+        }, 16);
```

Comment:
> 16

magic number?
Yes, i know that it is 1000/60
maybe deserve a comment.

Reply:
You're right.